scripts/Common.ps1

```diff
@@ -385,6 +385,202 @@ function Create-StorageContainer
+}
+
+## Function: New-AzureStorageContainer
+##
+## Purpose:
+##    Creates new storage container
+##
+## Input:
+##   ContainerName       name of the storage container
+##   AccessPolicy        Access policy (off,blob,container)
+##
+## Output:
+##   nothing
+##
+function New-AzureStorageContainer
+{
+
+    param(
+            [Parameter(Mandatory=$true)][string]$ContainerName,
+            [Parameter(Mandatory=$true)][string]$AccessPolicy
+         )
+
+    if ($AzureCliVersion -eq "1" )
+    {
+        # Azure Cli 1.0
+
+        # create the container now
+        $response = azure storage container create --account-name $StorageAccountName --account-key $StorageAccountKey --container $ContainerName --permission $AccessPolicy --json
+        $status = ((($response | jq --raw-output '.name') -ieq $ContainerName) -and (($response | jq --raw-output '.lease.state') -ieq 'available'))
+    }
+    else
+    {
+        # Azure Cli 2.0
+
+        # create the container now
+        if ($AzureStorageConnectionString)
+        {
+            Log-Message "Using connection string: $AzureStorageConnectionString" -Context "Create Storage Containers" -NoNewLine
+
+            $response = az storage container create --account-name $StorageAccountName --account-key $StorageAccountKey --name $ContainerName --public-access $AccessPolicy --connection-string $AzureStorageConnectionString -o json
+        }
+        else
+        {
+            $response = az storage container create --account-name $StorageAccountName --account-key $StorageAccountKey --name $ContainerName --public-access $AccessPolicy -o json
+        }
+
+        # parse the status (.created=true is the expected status)
+        $status = (($response | jq --raw-output '.created') -ieq "true")
+    }
+
+    # we expect the following: true=container created, If there is an error, status=[Blank]
+    if (!$status)
+    {
+        # creation failed
+        Log-Message "Unable to create the specified storage container: $ContainerName" -Context "Create Storage Containers"
+        exit 1
+    }
+}
+
+## Function: New-AzureStorageContainers
+##
+## Purpose:
+##    Create containers or modify existing container w.r.to existing permissions.
+##
+## Input:
+##   ContainerNames       name of the storage container
+##   AccessPolicy         Access policy (off,blob,container)
+##
+## Output:
+##   nothing
+##
+function New-AzureStorageContainers
+{
+    param(
+            [Parameter(Mandatory=$true)][string]$ContainerNames,
+            [Parameter(Mandatory=$true)][ValidateSet("off","blob","container")][string]$AccessPolicy
+         )
+
+    [array]$storageContainerList = $ContainerNames.Split(",");
+
+    foreach($storageContainerName in $storageContainerList)
+    {
+        $status = Get-AzureStorageContainerStatus -ContainerName $storageContainerName
+
+        if (!$status)
+        {
+            # Create the container if missing
+            Log-Message "'Creating Container' $storageContainerName"
```

Comment:
nit: move the single quote to the $storageContainerName parameter

Reply:
Done